File: 05/main.pl
use v5.38;

# Get lines from file
my @lines;
open my $fh, "<", "input.txt"; while (<$fh>) {chomp; push(@lines, $_)} close $fh;

my $nice1 = 0;
for (@lines) { $nice1++ if (m/(?:[aeiou].*){3,}/ && m/(\w)\1+/ && !m/ab|cd|pq|xy/)}
say "Part 1: $nice1";

my $nice2 = 0;
for (@lines) { $nice2++ if (m/(.{2}).*?\1/ && m/(.)(.)\1/)}
say "Part 2: $nice2";

File: 05/main.js
const fs = require('node:fs');

// Part1 checkers
const r1 = /(?:[aeiou].*){3,}/; // Minimum 3 vowels (a, e, i, o, u)
const r2 = /(\w)\1+/;           // Contains pairs of characters like "aa" or "xx"
const r3 = /ab|cd|pq|xy/;       // Contains forbidden words (ab, cd, pq, or xy)

// Part2 checkers
const r4 = /(.{2}).*?\1/        // Pair of chars that appear again, like "abab" or "abcab"
const r5 = /(.)(.)\1/           // Matches repeating letter after once char, like "xyx" or "efe"


fs.readFile('input.txt', 'utf8', (err, data) => {
    const list = data.split("\n");

    let nice1 = 0;
    for (const l of list) { if (r1.test(l) && r2.test(l) && !r3.test(l)) nice1++; }
    console.log(`Part1: ${nice1}`);

    let nice2 = 0;
    for (const l of list) { if (r4.test(l) && r5.test(l)) nice2++; }
    console.log(`Part2: ${nice2}`);
});